delfin/drivers/dell_emc/vmax/rest.py

```diff
@@ -371,6 +375,33 @@ def get_resource(self, array, category, resource_type,
             private=private, version=version)
         return self.get_request(target_uri, resource_type, params)
 
+    def get_resource_kwargs(self, *args, **kwargs):
+        """Get resource details from the array.
+
+        :key version: Unisphere version -- int
+        :key no_version: if versionless uri -- bool
+        :key category: the resource category e.g. sloprovisioning
+        :key resource_level: resource level e.g. storagegroup
+        :key resource_level_id: resource level id
+        :key resource_type: optional resource type e.g. maskingview
+        :key resource_type_id: optional resource type id
+        :key resource: the name of a specific resource
+        :key resource_id: the name of a specific resource
+        :key object_type: optional name of resource
+        :key object_type_id: optional name of resource
+        :key params: query parameters  -- dict
+        :key private: empty string or '/private' if private url
+        :returns: resource object -- dict or None
+        """
+        resource_type = None
+        if args:
+            resource_type = args[2]
+        elif not args and kwargs:
```

Comment:
When comes to this `elif`, it means that the first `if` in line 397 not match, so I think there is no need to add the condition `not args` here.

Reply:
Done